lib/hdk_time_indexing/src/index_tree.rs

```diff
@@ -144,14 +144,14 @@ pub (crate) fn get_index_segments(time: &DateTime<Utc>) -> Vec<IndexSegment> {
 /// Returns a `TimeIndexingError::Malformed` if an invalid link tag is passed.
 ///
 fn decode_link_tag_timestamp(tag: LinkTag) -> TimeIndexResult<DateTime<Utc>> {
-    let bits = tag.as_ref().split(|byte| { *byte == 0x0 as u8 });
+    let bits: Vec<&[u8]> = tag.as_ref().splitn(2, |byte| { *byte == 0x0 as u8 }).collect();
```

Comment:
I have a hard time reading the code on this line, as it uses some expressions and syntaxes I'm not familiar with. Could you put a code comment on it to say a snippet about what this does?